src/Core/Path/PathautoGeneratorTrait.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace drunomics\ServiceUtils\Core\Path;
+
+use Drupal\pathauto\PathautoGeneratorInterface;
+
+/**
+ * Allows setter injection and simple usage of the pathauto.generator service.
+ */
+trait PathautoGeneratorTrait {
+
+  /**
+   * The pathauto generator service.
+   *
+   * @var \Drupal\pathauto\PathautoGeneratorInterface
+   */
+  protected $pathautoGenerator;
+
+  /**
+   * Sets the pathauto generator service.
+   *
+   * @param \Drupal\pathauto\PathautoGeneratorInterface $pathautoGenerator
+   *   The pathauto generator service.
+   *
+   * @return $this
+   */
+  public function setPathautoGenerator(PathautoGeneratorInterface $pathautoGenerator) {
+    $this->pathautoGenerator = $pathautoGenerator;
+    return $this;
+  }
+
+  /**
+   * Gets the pathauto generator service.
+   *
+   * @return \Drupal\pathauto\PathautoGeneratorInterface
+   *   The pathauto generator service.
+   */
+  public function getPathautoGenerator() {
```

Comment:
looks good, but every trait also needs a respective test case added. simply copy over an existing test and adapt.